### PR TITLE
Fix gcc9 warning in EventFilter/SiStripRawToDigi

### PR DESCRIPTION
**Pull request by mrodozov**
#### PR description:

Fix gcc9 warning in EventFilter/SiStripRawToDigi

#### PR validation:
Builds without the warning

### CODE REVIEW

**Comment by mrodozov** [comment]
please test

File: EventFilter/SiStripRawToDigi/src/SiStripFEDBufferComponents.cc [hunk 1]
```diff
@@ -845,6 +845,7 @@ namespace sistrip {
       case READOUT_MODE_SCOPE:
         //scope mode is always real
         setReadoutModeBits(readoutMode);
+        [[fallthrough]];
```
Comment (perrotta) [inline EventFilter/SiStripRawToDigi/src/SiStripFEDBufferComponents.cc]: It looks like there is some odd logic behind...

As it is now, all case's but the default one foresee
```C++
    setReadoutModeBits(readoutMode);
```
and therefore one could simplier remove L846, L847, L848, L862, L863 (and at this point even L865).

Could you please confirm that this is indeed the intended behaviour @OlivierBondu @alesaggio ?

Comment (mmusich) [reply EventFilter/SiStripRawToDigi/src/SiStripFEDBufferComponents.cc]: @pieterdavid FYI

@perrotta, Olivier has left CMS, please update the list of reco contact for strips accordingly.

Comment (perrotta) [reply EventFilter/SiStripRawToDigi/src/SiStripFEDBufferComponents.cc]: > @pieterdavid FYI
> 
> @perrotta, Olivier has left CMS, please update the list of reco contact for strips accordingly.

Done in https://twiki.cern.ch/twiki/bin/view/CMS/RecoContacts

Comment (pieterdavid) [reply EventFilter/SiStripRawToDigi/src/SiStripFEDBufferComponents.cc]: This simplification should be correct and safe indeed.

**Comment by perrotta** [comment]
@mrodozov , given https://github.com/cms-sw/cmssw/pull/27846#discussion_r317352647 and https://github.com/cms-sw/cmssw/pull/27846#discussion_r317352647, please update this PR by simply removing L846-847-848, L862-863, L865

**Comment by mrodozov** [comment]
please test

**Comment by perrotta** [comment]
+1
- Following the issue pointed out by the gcc9 warning, the code in the switch was updated as agreed with the strip reco contacts
- Jenkins tests pass and show no differences

**Comment by kpedro88** [comment]
+1

**Pull request merged**